metabench.cmake

```diff
@@ -170,80 +194,112 @@ endfunction()
+"      # did not have the time to see the changes to the measure file and                               \n"
+"      # the target was not rebuilt. So we sleep for a bit and then retry                               \n"
+"      # this iteration.                                                                                \n"
+"      (sleep 0.2; redo) if command_line.nil?                                                           \n"
+"      error_string = <<-EOS                                                                            \n"
+"compilation error: #{command_line.captures[0]}                                                         \n"
+"                                                                                                       \n"
+"stdout                                                                                                 \n"
+"#{'-'*80}                                                                                              \n"
+"#{stdout}                                                                                              \n"
+"                                                                                                       \n"
+"stderr                                                                                                 \n"
+"#{'-'*80}                                                                                              \n"
+"#{stderr}                                                                                              \n"
+"                                                                                                       \n"
+"code                                                                                                   \n"
+"#{'-'*80}                                                                                              \n"
+"#{code}                                                                                                \n"
 "EOS\n"
-"    raise error_string if not status.success?                               \n"
-"  }                                                                         \n"
-"  datum[:values] << [n, time]                                               \n"
-"  progress.increment                                                        \n"
-"  source_file.write('')                                                     \n"
-"end                                                                         \n"
-"IO.write('\@datum\@', datum.to_json)                                        \n"
+"      raise error_string if not status.success?                                                        \n"
+"      data[:time] = stdout.match(/\\[compilation time: (.+)\\]/i).captures[0].to_f                     \n"
+"      # Size of the generated executable in KB                                                         \n"
+"      data[:size] = File.size(exe_file).to_f / 1000                                                    \n"
```

Comment:
shouldn't we express the executable size in base two rather than base ten, so this would be 1024?